src/gurt/hash.c

```diff
@@ -238,9 +238,25 @@ ch_bucket_lock(struct d_hash_table *htable, uint32_t idx, bool read_only)
 	if (htable->ht_feats & D_HASH_FT_MUTEX) {
 		D_MUTEX_LOCK(&lock->mutex);
 	} else if (htable->ht_feats & D_HASH_FT_RWLOCK) {
-		if (read_only)
+		if (read_only) {
+			uint32_t count;
+			uint32_t old_max;
+			bool     updated = false;
+
 			D_RWLOCK_RDLOCK(&lock->rwlock);
-		else
+			count = atomic_fetch_add_relaxed(
+			    &htable->ht_buckets[idx].hb_read_lock_count, 1);
+			do {
+				old_max = atomic_load_relaxed(
```

Comment:
i guess these lock counter is for debugging purpose? we shall disable it in normal case, it might affect performance..

Reply:
Yes, these are just for debugging purpose, I intend to update PR to compile these out by default.